weblate/memory/machine.py

```diff
@@ -72,4 +74,7 @@ def download_translations(
                 "origin": result.get_origin_display(),
                 "source": result.source,
                 "show_quality": True,
+                "delete_url": reverse("api:memory-detail", kwargs={"pk": result.id})
+                if user.has_perm("memory.delete", result)
```

Comment:
User can be None in some cases (for example when invoked via automatic translation add-on):

```suggestion
                if user is not None and user.has_perm("memory.delete", result)
```

See https://github.com/WeblateOrg/weblate/actions/runs/3083360977/jobs/4984199844#step:14:5298:

> weblate.machinery.base.MachineTranslationError: AttributeError: 'NoneType' object has no attribute 'has_perm'